stan/math/rev/fun/from_var_value.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef STAN_MATH_REV_FUN_FROM_VAR_VALUE_HPP
+#define STAN_MATH_REV_FUN_FROM_VAR_VALUE_HPP
+
+#include <stan/math/rev/core.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/rev/functor/arena_matrix.hpp>
+#include <stan/math/rev/functor/reverse_pass_callback.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Converts `var_value` into an Eigen Matrix. Adjoint is propagated back to
+ * arguument in the reverse pass.
+ *
+ * @tparam T type of the input
+ * @param a matrix to convert
+ */
+template <typename T, require_var_vt<is_eigen, T>* = nullptr>
+Eigen::Matrix<var, value_type_t<T>::RowsAtCompileTime,
+              value_type_t<T>::ColsAtCompileTime>
```

Comment:
`Rows/ColsAtCompileTime` are available in `vari_value`. Let's add them to `var_value` so then we can remove the value_type_t here.

I'm pretty sure all we need to do is add a `static constexpr int Rows/ColsAtCompileTime{0}` to the `vari_value<double>` class then the `var_value` can just use

```cpp
static constexpr int RowsAtCompileTime{vari_type::RowsAtCompileTime};
static constexpr int ColsAtCompileTime{vari_type::ColsAtCompileTime};
```

Reply:
Done.